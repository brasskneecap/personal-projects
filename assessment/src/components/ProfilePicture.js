const ProfilePicture = ({imgLocation}) => {
    return (
        <div>
            <div>
                <img className="picture-formatter" src={imgLocation} />
            </div>
        </div>
    )
}

export default ProfilePicture