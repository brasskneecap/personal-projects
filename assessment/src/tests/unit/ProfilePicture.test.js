import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ProfilePicture from '../../components/ProfilePicture';
import picture from "../../assets/Terran.jpg"

describe('ProfilePicture', () => {
    const wrapper = new ShallowRenderer()

    wrapper.render(
        <ProfilePicture imgLocation={picture} />
    );
    
    let componentRendered = wrapper.getRenderOutput();
    
    it('Should match ProfilePicture Snapshot', () => {
        expect(componentRendered).toMatchSnapshot();
    })

    it('Should contain image', () => {
        expect(componentRendered.props.children.props.children.props.src).toBe('Terran.jpg')
    })
})