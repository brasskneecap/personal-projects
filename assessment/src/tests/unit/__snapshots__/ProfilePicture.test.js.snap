// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePicture Should match ProfilePicture Snapshot 1`] = `
<div>
  <div>
    <img
      className="picture-formatter"
      src="Terran.jpg"
    />
  </div>
</div>
`;
